x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Types.java

```diff
@@ -20,16 +23,18 @@
     private Types() {}
 
     private static final Set<String> CATEGORICAL_TYPES =
-        Collections.unmodifiableSet(
-            Stream.of("text", "keyword", "ip")
-                .collect(Collectors.toSet()));
+        Stream.of(TextFieldMapper.CONTENT_TYPE, KeywordFieldMapper.CONTENT_TYPE, IpFieldMapper.CONTENT_TYPE)
+            .collect(Collectors.toUnmodifiableSet());
 
     private static final Set<String> NUMERICAL_TYPES =
-        Collections.unmodifiableSet(
-            Stream.concat(
-                    Stream.of(NumberFieldMapper.NumberType.values()).map(NumberFieldMapper.NumberType::typeName),
-                    Stream.of("scaled_float"))
-                .collect(Collectors.toSet()));
+        Stream.concat(Stream.of(NumberType.values()).map(NumberType::typeName), Stream.of("scaled_float"))
+            .collect(Collectors.toUnmodifiableSet());
+
+    private static final Set<String> BOOLEAN_AND_DISCRETE_NUMERICAL_TYPES =
```

Comment:
Since we can mix 'n match, why not have boolean and discrete_numerical_types separately? Then calling code can mix 'n match flexibly

Reply:
Done.